cosmos/operators/_async/bigquery.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Sequence
+
+from airflow.providers.google.cloud.hooks.bigquery import BigQueryHook
+from airflow.providers.google.cloud.operators.bigquery import BigQueryInsertJobOperator
+from airflow.utils.context import Context
+
+from cosmos import settings
+from cosmos.config import ProfileConfig
+from cosmos.exceptions import CosmosValueError
+from cosmos.settings import remote_target_path, remote_target_path_conn_id
+
+
+class DbtRunAirflowAsyncBigqueryOperator(BigQueryInsertJobOperator):  # type: ignore[misc]
+
+    template_fields: Sequence[str] = (
+        "full_refresh",
+        "gcp_project",
+        "dataset",
+        "location",
+    )
+
+    def __init__(
+        self,
+        project_dir: str,
+        profile_config: ProfileConfig,
+        extra_context: dict[str, Any] | None = None,
+        async_args: dict[str, Any] | None = None,
+        **kwargs: Any,
+    ):
+        if async_args is None:
+            async_args = {}
+        self.project_dir = project_dir
+        self.profile_config = profile_config
+        self.gcp_conn_id = self.profile_config.profile_mapping.conn_id  # type: ignore
+        profile = self.profile_config.profile_mapping.profile  # type: ignore
+        self.gcp_project = profile["project"]
```

Comment:
I didn't realise how connected our async implementation may be to the profile implementation. 

In Cosmos 2.x, we may want to have something like:
- `cosmos/adaptors/bigquery`
     - `profiles/`
     - `operators/async.py`

Reply:
Yes, something like this makes sense, but for the scope of this PR, I'll keep the dir structure unchanged